packages/riipen-ui/src/components/Popover.jsx

```diff
@@ -0,0 +1,300 @@
+    window.addEventListener("scroll", positionChangeEventHandler);
+    window.addEventListener("mousedown", closeEventHandler);
+    window.addEventListener("keydown", closeEventHandler);
+
+    if (anchorEl && lockScroll) {
+      this.lockParentScroll();
+    }
+  }
+
+  componentWillUnmount() {
+    const { positionChangeEventHandler, closeEventHandler } = this.state;
+    const { anchorEl, lockScroll } = this.props;
+    window.removeEventListener("resize", positionChangeEventHandler);
+    window.removeEventListener("scroll", positionChangeEventHandler);
+    window.removeEventListener("mousedown", closeEventHandler);
+    window.removeEventListener("keydown", closeEventHandler);
+
+    if (anchorEl && lockScroll) {
+      this.resetParentScroll();
+    }
+  }
+
+  getAnchorEl = () => {
+    const { anchorEl } = this.props;
+    return typeof anchorEl === "function" ? anchorEl() : anchorEl;
+  };
+
+  getPositioningStyle = () => {
+    const { contentRef } = this.state;
+    const {
+      anchorPosition = {
+        vertical: "top",
+        horizontal: "left"
+      },
+      contentPosition = {
+        vertical: "top",
+        horizontal: "left"
+      }
+    } = this.props;
+
+    const anchorEl = this.getAnchorEl();
+    if (!anchorEl || !contentRef) return {};
+
+    // Set top and left of element based on location of anchor
+    const anchorRect = anchorEl.getBoundingClientRect();
+    let top = anchorRect.top;
+    let left = anchorRect.left;
+
+    // Offset Content Based on anchorPosition props
+    const anchorVerticalOffset = getOffsetTop(
+      anchorRect,
+      anchorPosition.vertical
+    );
+    const anchorHorizonalOffset = getOffsetLeft(
+      anchorRect,
+      anchorPosition.horizontal
+    );
+
+    top += anchorVerticalOffset;
+    left += anchorHorizonalOffset;
+
+    // Offset Content Based on contentPosition props
+    const contentRect = contentRef.getBoundingClientRect();
+
+    const contentVerticalOffset = getOffsetTop(
+      contentRect,
+      contentPosition.vertical
+    );
+    const contentHorizontalOffset = getOffsetLeft(
+      contentRect,
+      contentPosition.horizontal
+    );
+
+    top -= contentVerticalOffset;
+    left -= contentHorizontalOffset;
+
+    // Move menu back into view if out of screen
+    const { marginThreshold = 0 } = this.props;
+    const viewContainer = getContainer(anchorEl);
+
+    const heightMax = viewContainer.innerHeight - marginThreshold;
+    const widthMax = viewContainer.innerWidth - marginThreshold;
+
+    // Check Vertical Constraints
+    if (top + contentRect.height > heightMax) {
+      top -= top + contentRect.height - heightMax;
+    } else if (top < marginThreshold) {
+      top = marginThreshold;
+    }
+
+    // Check Horizontal Constraints
+    if (left + contentRect.width > widthMax) {
+      left -= left + contentRect.width - widthMax;
+    } else if (left < marginThreshold) {
+      left = marginThreshold;
+    }
```

Comment:
I am not sure if this is necessary, This is the part that adjusts the popup to make sure its always on screen. Im just am thinking that in the future you may want to scroll an input off screen and it should actually move off screen when you do that. Maybe just adjust the position for the content in the container?

Reply:
I think if the anchor element scrolls off screen then the menu should too?

Reply:
Okay, Ill make the change and then play around with it and see how it feels.